internal/opnsense-unbound/client.go

```diff
@@ -8,26 +8,37 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"net/url"
+	"path"
+	"strings"
 
 	log "github.com/sirupsen/logrus"
 	"sigs.k8s.io/external-dns/endpoint"
 )
 
+const emptyJSONObject = "{}"
+
 // httpClient is the DNS provider client.
 type httpClient struct {
 	*Config
 	*http.Client
+	baseURL *url.URL
 }
 
 const (
-	opnsenseUnboundServicePath  = "%s/api/unbound/service/%s"
-	opnsenseUnboundSettingsPath = "%s/api/unbound/settings/%s"
+	opnsenseUnboundServicePath  = "service"
+	opnsenseUnboundSettingsPath = "settings"
 	// Hacky, but nice to have the delete as an explicit constant since it's destructive
-	opnsenseUnboundSettingsPathDelete = "%s/api/unbound/settings/delHostOverride/%s"
+	opnsenseUnboundSettingsPathDelete = "settings/delHostOverride"
 )
```

Comment:
I wonder if the constants should be kept? Is there value if it's not required for string formatting?

Reply:
they're only used for constructing the URL strings. my guess is that if they can be removed while preserving the functionality of the URL construction in the subsequent functions...we can just ditch them?

Reply:
Okay, will update it.